jcl/src/java.base/share/classes/java/lang/StringBuilder.java

```diff
@@ -208,7 +209,7 @@ public StringBuilder append (char[] chars) {
 		} else {
 			// Check if the StringBuilder is compressed
 			if (count >= 0) {
-				decompress(newLength);
+				currentCapacity = decompress(newLength) & ~sharedBit;
```

Comment:
Why do we need to mask off the shared bit? `decompress` should always allocate a new array, meaning the array should never be shared at that point.

Reply:
Updated the PR to removing masking this bit